### ISSUE TITLE
[QST] Keep Getting Error: "The labels need to be consecutive values from 0 to the number of unique label values"

### ISSUE THREAD

**Issue by mexicantexan** [rapidsai/cudf#10024]
**What is your question?**
I'm trying to fit data to the cuml.ensemble.RandomForestClassifier and I keep getting the error: "The labels need to be consecutive values from 0 to the number of unique label values"

I'm passing cudf.DataFrame objects into the function which have the same number of rows but differing number of columns. The column labels start at 0 and step by 1 up to the final column. What am I doing wrong? I've attached a printout of the dataframes that I'm passing in below and some code for context:

<code>clf1 = modelClass(max_depth=D1, random_state=random.randrange(0, 1024, 1), 
     n_bins=15, n_streams=4, split_criterion=criterion, bootstrap=bootstrap, n_estimators=trs1)</code>
<code>clf1.fit(X1, Y1)</code>
X1's dataframe looks like this:
                 0             1                        2                 ...       106        107        108
0     1.000000e-11  1.000000e-11  1.647421e-01  ...  6.416093   6.905203   0.159865
1     1.000000e-11  1.000000e-11  1.760000e-02  ...  0.000000 -25.956954 -21.199595
2     1.000000e-11  1.000000e-11 -1.772000e-01  ... -4.354171  -1.810521   2.681238
3     1.000000e-11  1.000000e-11  8.254000e-01  ...  0.000000  -2.439157  -3.359210
4     1.000000e-11  1.000000e-11  2.587000e-01  ...  2.091000   2.132047   1.682897
...            ...           ...           ...  ...       ...        ...        ...
5402  1.000000e-11  1.000000e-11  1.704444e-01  ...  0.240028   0.902110   1.267209
5403  1.000000e-11  1.000000e-11 -1.860000e-01  ...  1.142419   3.116908   2.839310
5404  0.000000e+00  1.000000e-11  1.229714e-01  ... -1.558528  -2.459344  -1.862828
5405  1.000000e-11  1.959500e-01  1.984667e-01  ...  0.000000   2.590066   2.050400
5406  1.000000e-11  1.000000e-11  1.000000e-11  ... -3.435832  -1.341034   7...
[5407 rows x 109 columns]; dtype=('0', dtype('float64')); <cudf.core.dataframe._DataFrameLocIndexer object at 0x7f9c3d0f3070>

Y1's Dataframe looks like this:
        0
0    -2.0
1     4.0
2    -3.0
3     1.0
4     0.0
...   ...
5402  0.0
5403 -2.0
5404  0.0
5405  0.0
5406  0.0

[5407 rows x 1 columns]; dtype=('0', dtype('float64')); <cudf.core.dataframe._DataFrameLocIndexer object at 0x7f9c1b847b50>

System Information: Ubuntu 20.04, Titan RTX, CUDA 11.5, Rapids 21.12 built in Conda, Python 3.8

**Comment by venkywonka** [comment]
Hi @mexicantexan , thanks for the question.  The "labels" in question are the class labels of the Y1 dataframe. The `RandomForestClassifier` requires the Y class labels to be whole numbers starting from 0. In your case, seems like it starts from some negative number (-3.0?). A naive way to make it conformant is by just elementwise adding MIN_LABEL to bring them to the range [0, num_unique_class_labels].

**Comment by beckernick** [comment]
Note that with non-consecutive labels outside of the `[0, n)` range you will not experience this error in scikit-learn, perhaps due to label encoding happening under the hood.

```python
import pandas as pd
from sklearn.ensemble import RandomForestClassifier
​
df = pd.DataFrame({
    "x1": [0,1,2],
    "x2": [-3,2,5],
    "y": [-3, 0, 4.0]
})
clf = RandomForestClassifier()
clf.fit(df[["x1", "x2"]], df["y"])
RandomForestClassifier()
```

I'm going to close this issue as answered, and recommend filing an issue on [cuML issue board](https://github.com/rapidsai/cuml/issues) if this behavior is desired in cuML.

**Comment by beckernick** [comment]
Filed https://github.com/rapidsai/cuml/issues/4478 and closing

**Comment by mexicantexan** [comment]
@beckernick I appreciate it! My fault I figured I was misusing the cuDF DataFrames, so asked it here naively.
